### PR TITLE
feat(schema-tools): support name space on schema creation

### LINKED ISSUE #8038: Name spaces not supported when creating a Schema instance from a given set of metadata classes
### Bug Report

|    Q        |   A
|------------ | ------
| BC Break    | no
| Version     | 2.7.1

#### Current behavior
Name spaces not supported during schema creation.

#### How to reproduce

1.  Create an entity with `@Table(schema="mySchema", name="myTable")`
2. Launch the command `doctrine:schema:update --force`
3. you'll have an SQL error `the schema mySchema doesn't exists`
4. To understand exactly what happen, lunch the command `doctrine:schema:update --dump-sql`, you'll see that an SQL instruction is missing `create schema mySchema;`

#### Expected behavior
Support name spaces during schema creation.

### PR DESCRIPTION
**Pull request by abdel-geniusespro**
Closes #8038

### CODE REVIEW

File: lib/Doctrine/ORM/Tools/SchemaTool.php [hunk 1]
```diff
@@ -164,6 +168,7 @@ public function getSchemaFromMetadata(array $classes)
             }
 
             $table = $schema->createTable($this->quoteStrategy->getTableName($class, $this->platform));
+            $this->addSchemaNameSpaceFromTable($schema, $table);
```
Comment (greg0ire) [inline lib/Doctrine/ORM/Tools/SchemaTool.php]:
```suggestion
            $this->addSchemaNamespaceFromTable($schema, $table);
```

File: lib/Doctrine/ORM/Tools/SchemaTool.php [hunk 2]
```diff
@@ -368,6 +373,27 @@ function (ClassMetadata $class) use ($idMapping) : bool {
...
     }
 
+    /**
+     * @param Schema $schema
+     * @param Table  $table
+     *
+     * @throws DBALException
+     * @throws SchemaException
+     */
+    private function addSchemaNameSpaceFromTable(Schema $schema, Table $table): void
```
Comment (greg0ire) [inline lib/Doctrine/ORM/Tools/SchemaTool.php]:
```suggestion
    private function addSchemaNamespaceFromTable(Schema $schema, Table $table): void
```

File: lib/Doctrine/ORM/Tools/SchemaTool.php [hunk 3]
```diff
@@ -368,6 +373,27 @@ function (ClassMetadata $class) use ($idMapping) : bool {
         return $schema;
     }
 
+    /**
+     * @param Schema $schema
+     * @param Table  $table
+     *
```
Comment (greg0ire) [inline lib/Doctrine/ORM/Tools/SchemaTool.php]: No need for such phpdoc when you have type declarations.

```suggestion
```

**Review by greg0ire — COMMENTED** [review]
There probably should be tests for this, and maybe docs.

File: lib/Doctrine/ORM/Tools/SchemaTool.php [hunk 4]
```diff
@@ -141,7 +143,9 @@ private function processingNotRequired($class, array $processedClasses)
      *
      * @return Schema
      *
-     * @throws \Doctrine\ORM\ORMException
+     * @throws ORMException
+     * @throws DBALException
+     * @throws SchemaException
```
Comment (SenseException) [inline lib/Doctrine/ORM/Tools/SchemaTool.php]: `DBALException`and `SchemaException` are part of DBAL. Are those exceptions caught in ORM by using this method? If not, I would expect that an `ORMException` would be thrown if internally the DBAL exceptions occur.

File: lib/Doctrine/ORM/Tools/SchemaTool.php (hunk 3 above)
Comment (abdel-geniusespro) [reply lib/Doctrine/ORM/Tools/SchemaTool.php]: I did it to keep the file homogeneous

**Comment by abdel-geniusespro** [comment]
Thank's @SenseException & @greg0ire for the review, but this is not the best way to solve this problem.

**Pull request closed without merging**